supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/monitoring/CheckpointingConfiguration.scala

```diff
@@ -0,0 +1,74 @@
+package cromwell.backend.google.pipelines.common.monitoring
+
+import cromwell.backend.BackendJobDescriptor
+import cromwell.backend.io.WorkflowPaths
+import cromwell.core.path.Path
+
+import scala.concurrent.duration.FiniteDuration
+
+final class CheckpointingConfiguration(jobDescriptor: BackendJobDescriptor,
+                                       workflowPaths: WorkflowPaths,
+                                       commandDirectory: Path,
+                                       checkpointInterval: FiniteDuration
+                                      ) {
+  def checkpointFileCloud(checkpointFileName: String): String = {
+    // Fix the attempt at 1 because we always use the base directory to store the checkpoint file.
```

Comment:
So when a VM is preempted during attempt-1, we copy the contents of the checkpoint file to another file in path `../attempt-2/_checkpointing/my_checkpoint` and continue progressing on the task and adding checkpoints in there. But what happens if attempt-2 VM is also preempted? During the attempt-3 will Cromwell look for checkpoints in the base directory or look at the checkpoint file inside attempt-2 checkpoint i.e. `.../attempt-2/_checkpointing/my_checkpoint` file?

Reply:
Discussed in person. I'll update the comment to hopefully be more clear